### ISSUE TITLE
Consuming a ParallelIterator with a for loop

### ISSUE THREAD

**Issue by kpcyrd** [rayon-rs/rayon#645]
It seems all rayon examples use something functional like `.sum()` or `for_each` with a function that can be executed in parallel.

Sometimes I want to run a `.map()` in parallel for each item but consume the results sequentially with a for loop. Is there a way to do that?

I'm currently back to rolling my own with the threadpool crate.

**Comment by cuviper** [comment]
We don't have this in rayon yet -- #210.

In the meantime, you might like the [`par-map`](https://crates.io/crates/par-map) crate as an alternative.

**Comment by nikomatsakis** [comment]
Closing as a duplicate of #210 --thanks @kpcyrd !

**Comment by safinaskar** [comment]
Here is my solution: https://github.com/rayon-rs/rayon/pull/1071
